onnxruntime/test/testdata/custom_op_library/custom_op_library.cc

```diff
@@ -73,6 +79,45 @@ struct KernelOne {
   Ort::CustomOpApi ort_;
 };
 
+struct KernelOneCuda {
+  KernelOneCuda(OrtApi api)
+      : api_(api),
+        ort_(api_) {
+  }
+
+  void Compute(OrtKernelContext* context) {
+    // Setup inputs
+    const OrtValue* input_X = ort_.KernelContext_GetInput(context, 0);
+    const OrtValue* input_Y = ort_.KernelContext_GetInput(context, 1);
+    const float* X = ort_.GetTensorData<float>(input_X);
+    const float* Y = ort_.GetTensorData<float>(input_Y);
+
+    // Setup output
+    OrtTensorDimensions dimensions(ort_, input_X);
+
+    OrtValue* output = ort_.KernelContext_GetOutput(context, 0, dimensions.data(), dimensions.size());
+    float* out = ort_.GetTensorMutableData<float>(output);
+
+    OrtTensorTypeAndShapeInfo* output_info = ort_.GetTensorTypeAndShape(output);
+    int64_t size = ort_.GetTensorShapeElementCount(output_info);
+    ort_.ReleaseTensorTypeAndShapeInfo(output_info);
+
+    // Do computation
+#ifdef USE_CUDA
+    cuda_add(size, out, X, Y, 0);
+    cudaDeviceSynchronize();
```

Comment:
I think we should avoid adding this costly sync in code that may potentially be looked at as a sample by users.

This sync is to be added only if the custom op is launched in a different cuda stream to the one used in the session that runs the model using this custom kernel. See https://github.com/microsoft/onnxruntime/pull/6971 for more context.

For custom op shared library, removing the cudaDeviceSynch is going to get ugly. I can think of only one way: 
1) Create a cuda stream in the shared library (which can be destroyed when the library is unloaded). 
2) Export a method in the shared library (like `RegisterCustomOps()` is exported in the shared library) to query the created cuda stream and use the very stream to launch a session that is going to use the cuda custom kernel.

It is ugly but it should work.

@weixingzhang was looking into supporting an API to query the cuda kernel used by an ort session. I am not sure if it will help the case of pre-compiled shared libraries containing custom ops. It needs some more thought...